app.rb

```diff
@@ -0,0 +1,97 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  WORDS_NUMBERS_SPACES_REGEX = /^[\w\s]+$/.freeze
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    puts(@books.map { |book| "Title: \"#{book.title}\", Author: #{book.author}" })
+  end
+
+  def list_people
+    puts(@people.map { |person| "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" })
+  end
+
+  def create_person
+    option = input('number', 1..2, 'Do you want to create a student (1) or a teacher (2)? [Input the number]: ')
+    age = input('number', 1..1000, 'Age: ')
+    name = input('string', WORDS_NUMBERS_SPACES_REGEX, 'Name: ')
+    case option
+    when 1
+      parent_permission = input('string', /^[ynYN]$/, 'Has parent permission? [Y/N]: ')
+      @people.push(Student.new(age, name, parent_permission: parent_permission.downcase == 'y'))
+    when 2
+      specialization = input('string', WORDS_NUMBERS_SPACES_REGEX, 'Specialization: ')
+      @people.push(Teacher.new(specialization, age, name))
+    end
+    puts 'Person created successfully'
+  end
+
+  def create_book
+    title = input('string', WORDS_NUMBERS_SPACES_REGEX, 'Title: ')
+    author = input('string', WORDS_NUMBERS_SPACES_REGEX, 'Author: ')
+    @books.push(Book.new(title, author))
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    print_text = "Select a book from the following list by number\n"
+    print_text += @books.map.with_index do |book, index|
+      "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end.join("\n").concat("\n")
+    book_index = input('number', 0..(@books.length - 1), print_text)
+    puts ''
+    print_text = "Select a person from the following list by number (not id)\n"
+    print_text += @people.map.with_index do |person, index|
+      "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end.join("\n").concat("\n")
+    person_index = input('number', 0..(@people.length - 1), print_text)
+    puts ''
+    date = input('string', %r{^\d{4}/\d{2}/\d{2}$}, 'Date: ')
+    @rentals.push(Rental.new(date, @books[book_index], @people[person_index]))
+    puts 'Rental created successfully'
```

Comment:
## Suggested Change

- [ ] To make your app more user-friendly, it would be nice if you could place a date guide for a user when entering a date. For instance, you can place a date format - Date:(2023/05/15) to guide the user in entering the right date format. 👍🏽